sdk/typescript/src/providers/provider.ts

```diff
@@ -119,5 +122,18 @@ export abstract class Provider {
   ): Promise<SuiMoveNormalizedStruct>;
 
   abstract syncAccountState(address: string): Promise<any>;
+
+  /**
+   * Subscribe to an event
+   */
+   abstract subscribeEvent(
+    filter: SuiEventFilter,
+    onMessage: (event: SuiEventEnvelope) => void
+  ): Promise<SubscriptionId>;
+
+  /**
+   * Unsubscribe from an event
+   */
+  abstract unsubscribeEvent(id: SubscriptionId): Promise<boolean>;
```

Comment:
Can we add more detailed usage instructions here? We/I haven't been doing a great job in the function documentation and have recently received some feedback from the community for the lack of documentation in the SDK.

Reply:
added JSDoc `@param` information and a bit more text